app/javascript/react/screens/App/Overview/components/InfrastructureMappingsList/InfrastructureMappingsList.js

```diff
@@ -1,226 +1,551 @@
+
+              {error ? (
+                <OverviewEmptyState
+                  title={__('Error loading mappings.')}
+                  iconType="pf"
+                  iconName="error-circle-o"
+                  description={
+                    <React.Fragment>
+                      <span>
+                        {__(
+                          'There was an error loading Infrastructure Mappings.'
+                        )}
+                      </span>
+                      <br />
+                      <span>{__('Please refresh and try again.')}</span>
+                    </React.Fragment>
+                  }
+                />
+              ) : (
+                <ListView
+                  style={{ marginTop: 10 }}
+                  className="infra-mappings-list-view"
+                  id="infrastructure_mappings"
+                >
+                  {transformationMappingsMutable.map(mapping => {
+                    const sourceClusterCount = this.sourceCount(
+                      mapping.transformation_mapping_items,
+                      'emscluster'
+                    );
+                    const targetClusterCount = this.destinationCount(
+                      mapping.transformation_mapping_items,
+                      'emscluster'
+                    );
+                    const sourceDatastoreCount = this.sourceCount(
+                      mapping.transformation_mapping_items,
+                      'storage'
+                    );
+                    const targetDatastoreCount = this.destinationCount(
+                      mapping.transformation_mapping_items,
+                      'storage'
+                    );
+                    const sourceLanCount = this.sourceCount(
+                      mapping.transformation_mapping_items,
+                      'lan'
+                    );
+                    const targetLanCount = this.destinationCount(
+                      mapping.transformation_mapping_items,
+                      'lan'
+                    );
+                    const associatedPlansCount =
+                      mapping.service_templates &&
+                      mapping.service_templates.length;
+
+                    const {
+                      targetClusters,
+                      targetDatastores,
+                      targetNetworks
+                    } = mapInfrastructureMappings(
+                      mapping.transformation_mapping_items,
+                      clusters,
+                      datastores,
+                      networks
+                    );
+
+                    return (
+                      <ListView.Item
+                        key={mapping.id}
+                        heading={mapping.name}
+                        description={
+                          <small>
+                            {__('Completed: ')}
+                            {moment(mapping.created_at).format(
+                              'MMMM Do YYYY, h:mm a'
```

Comment:
we do this in infastructuremappinglist.js and planrequestdetailslist... might be time to make a constant 🤔

Reply:
sure - can add a common helper...